pydoctor/driver.py

```diff
@@ -316,17 +320,20 @@ def main(args=sys.argv[1:]):
             error("The system does not contain any code, did you "
                   "forget an --add-package?")
 
-        system.process()
 
         if system.options.projectname is None:
             name = '/'.join(ro.name for ro in system.rootobjects)
-            system.msg(
-                'warning',
-                'WARNING: guessing '+name+' for project name', thresh=-1)
+            system._warning(
+                current=None,
+                message='guessing {} for project name'.format(name),
```

Comment:
You can use f-strings in pydoctor.

Reply:
By the way, should this actually be a warning?

How likely is it that we guess the project name correctly? If it's unlikely, then it's good to warn, since the name is probably incorrect (but we have to put something there). If it's likely, then warning will either create noise in the logging or force the user to pass an additional command line option while auto-detection would have sufficed.

I think there are quite a few Python projects, especially tools and libraries, where the project name is equal to the top-level package name. In that case, the message should be at the "info" rather than "warning" level, which unfortunately would make this a poor choice to base the test case on.

Reply:
Regarding f-string: long time python2.5 user here :) ... still learning the latest things from py3.

But I will update it.

Maybe use the pyupgrade tool to automatically complain about these things.

This was previously a "WARNING" .... so I keep it as such, since it makes it easier to write the tests :)
 
But I still that it should still be a warning, as is best for users to explicitly set the project name.
pydoctor did its best to guess the name, but an user should double check it.
Also, some people might want different cases for the name.

But if you have a strong feeling against making it a warning, I can make it a normal INFO message and will update the tests. I not hard to generate a warning in another way.